src/main/java/io/github/moulberry/notenoughupdates/profileviewer/BasicPage.java

```diff
@@ -822,55 +825,89 @@ private void renderWeight(
-		);
-		if (mouseX > guiLeft + 63 - textWidth / 2 && mouseX < guiLeft + 63 + textWidth / 2) {
-			if (mouseY > guiTop + 12 && mouseY < guiTop + 12 + fr.FONT_HEIGHT) {
-				getInstance().tooltipToDisplay = new ArrayList<>();
-				getInstance()
-					.tooltipToDisplay.add(
-						EnumChatFormatting.GREEN +
-							"Skills: " +
-							EnumChatFormatting.GOLD +
-							GuiProfileViewer.numberFormat.format(roundToNearestInt(senitherWeight
-								.getSkillsWeight()
-								.getWeightStruct()
-								.getRaw()))
-					);
-				getInstance()
-					.tooltipToDisplay.add(
-						EnumChatFormatting.GREEN +
-							"Slayer: " +
-							EnumChatFormatting.GOLD +
-							GuiProfileViewer.numberFormat.format(roundToNearestInt(senitherWeight
-								.getSlayerWeight()
-								.getWeightStruct()
-								.getRaw()))
-					);
-				getInstance()
-					.tooltipToDisplay.add(
-						EnumChatFormatting.GREEN +
-							"Dungeons: " +
-							EnumChatFormatting.GOLD +
-							GuiProfileViewer.numberFormat.format(
-								roundToNearestInt(senitherWeight.getDungeonsWeight().getWeightStruct().getRaw())
-							)
-					);
+		if (weight > 0) {
+			Utils.drawStringCentered(
+				EnumChatFormatting.GREEN +
+					"Senither Weight: " +
+					EnumChatFormatting.GOLD +
+					GuiProfileViewer.numberFormat.format(roundToNearestInt(senitherWeight.getTotalWeight().getRaw())),
+				fr,
+				guiLeft + 63,
+				guiTop + 18,
+				true,
+				0
+			);
+
+			int textWidth = fr.getStringWidth(
+				"Senither Weight: " +
+					GuiProfileViewer.numberFormat.format(roundToNearestInt(senitherWeight.getTotalWeight().getRaw()))
+			);
+			if (mouseX > guiLeft + 63 - textWidth / 2 && mouseX < guiLeft + 63 + textWidth / 2) {
+				if (mouseY > guiTop + 12 && mouseY < guiTop + 12 + fr.FONT_HEIGHT) {
+					getInstance().tooltipToDisplay = new ArrayList<>();
+					getInstance()
+						.tooltipToDisplay.add(
+							EnumChatFormatting.GREEN +
+								"Skills: " +
+								EnumChatFormatting.GOLD +
+								GuiProfileViewer.numberFormat.format(roundToNearestInt(senitherWeight
+									.getSkillsWeight()
+									.getWeightStruct()
+									.getRaw()))
+						);
+					getInstance()
+						.tooltipToDisplay.add(
+							EnumChatFormatting.GREEN +
+								"Slayer: " +
+								EnumChatFormatting.GOLD +
+								GuiProfileViewer.numberFormat.format(roundToNearestInt(senitherWeight
+									.getSlayerWeight()
+									.getWeightStruct()
+									.getRaw()))
+						);
+					getInstance()
+						.tooltipToDisplay.add(
+							EnumChatFormatting.GREEN +
+								"Dungeons: " +
+								EnumChatFormatting.GOLD +
+								GuiProfileViewer.numberFormat.format(
+									roundToNearestInt(senitherWeight.getDungeonsWeight().getWeightStruct().getRaw())
+								)
+						);
+
+					if (NotEnoughUpdates.INSTANCE.config.profileViewer.useSoopyNetworth
+						&& profile.getSoopyWeightLeaderboardPosition() >= 0
+						&& profile.isProfileMaxSoopyNetworth(profileId)) {
```

Comment:
btw i believe the senither lb on my website tracks profile with most senither weight not nw

iirc all my lbs are highest nw profile except for nw leaderboard

Reply:
oh nvm I just double checked my server's code and its all based on highest weight profile (even nw lb pos)

Reply:
so should i change it to isProfileMaxSoopyWeight ?